earth_enterprise/rpms/buildSrc/src/main/groovy/org/opengee/shell/GeeCommandLine.groovy

```diff
@@ -0,0 +1,66 @@
+package org.opengee.shell
+
+class GeeCommandLine {
+    class ExpectedResult {
+        Integer exitCode = null
+        String stdOut = null
+        String stdErr = null
+
+        ExpectedResult(Integer exitCode, String stdOut, String stdErr) {
+            this.exitCode = exitCode
+            this.stdOut = stdOut
+            this.stdErr = stdErr
+        }
+    }
+
+    // Returns the standard output that results from executing a given command.
+    //     To pipe to standard input set `stdInAction` to a lambda that
+    // accepts the standard input writer as an parameter.
+    //     To suppress known command results (return `null`), use the
+    // `suppressedResults` parameter.
+    static def expand(
+        command, error_message_header, stdInAction = null,
+        Iterable<ExpectedResult> suppressedResults = []
+    ) {
+        def process = command.execute()
+        def stdoutBuffer = new StringBuilder()
+        def stderrBuffer = new StringBuilder()
+
+        if (stdInAction != null) {
+            process.withWriter stdInAction
+        }
+        process.waitForProcessOutput(stdoutBuffer, stderrBuffer)
+        def exitCode = process.waitFor()
+        suppressedResults.each {
+            if (
+                (it.exitCode == null || it.exitCode == exitCode) &&
+                (it.stdOut == null || it.stdOut == stdoutBuffer) &&
+                (it.stdErr == null || it.stdErr == stderrBuffer)
+            ) {
+                // Suppress result:
+                return null;
+            }
+        }
+        if (exitCode != 0) {
+            throw new IllegalArgumentException("""\
```

Comment:
Why IllegalArgumentException? Is that the only way there could be an issue running the command or just the most likely?

Reply:
No, that would be just for the case when the command couldn't be found.  I'll look for a better exception.